.github/workflows/deployment.yml

```diff
@@ -28,7 +28,17 @@ on:
         default: true
 
 jobs:
+  validate-tag-name:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate tag name format
+        run: |
+          if [[ ! "${{ inputs.tag_name }}" =~ ^v[0-9]+.[0-9]+.[0-9]+$]]; then
```

Comment:
Is this supposed to be a simple semantic version regex? Did you mean to escape the dots?

Reply:
Ah, nope. Those are supposed to match `.` literally, not any character. Good catch! Of course the regex validator I was using wouldn't catch this as `.` is still any character